forward/forward.go

```diff
@@ -169,13 +172,6 @@ func (f *Forwarder) cloneRequest(req *http.Request, u *url.URL) (*http.Request,
 	outReq.URL.Scheme = "http"
 	// We need to make sure the host is defined in the URL (not the actual URI)
 	outReq.URL.Host = req.Host
-	// Make sure we define an opaque URL, so the URI is just the path
-	//
-	// Note that EscapedPath() always return the URL-encoded path, so we have
-	// no way to know if the raw URL in req is encoded or not.  Go HTTP client
-	// always encode URL, so do major browsers. That means it's totally safe for
-	// Lantern usage and possibly others.
-	outReq.URL.Opaque = req.URL.EscapedPath()
```

Comment:
Oh sorry it's incorrect. [`req. RequestURI`](https://golang.org/pkg/net/http/#Request) does preserve the original URI, but the doc says `It is an error to set this field in an HTTP client request.`, so we need some trick to forward the original URI.

The reason we set `outReq.URL.Opaque` should related to below comments in `URL` package.

> Go 1.5 introduced the RawPath field to hold the encoded form of Path. The Parse function sets both Path and RawPath in the URL it returns, and URL's String method uses RawPath if it is a valid encoding of Path, by calling the EscapedPath method.

> In earlier versions of Go, the more indirect workarounds were that an HTTP server could consult req.RequestURI and an HTTP client could construct a URL struct directly and **set the Opaque field instead of Path**. These still work as well.